src/libutil/util.cc

```diff
@@ -1688,13 +1688,21 @@ void setStackSize(size_t stackSize)
     #endif
 }
 
+#if __linux__
 static AutoCloseFD fdSavedMountNamespace;
+std::optional<Path> savedCwd;
+#endif
 
 void saveMountNamespace()
 {
 #if __linux__
     static std::once_flag done;
     std::call_once(done, []() {
+        char* cwd = getcwd(NULL, 0);
```

Comment:
`getcwd` allocating its return value is a GNU extension, so maybe we shouldn't rely on that working

Reply:
How's [`2a45cf5` (#6348)](https://github.com/NixOS/nix/pull/6348/commits/2a45cf54e4201a894254604676dcb51f4c1a471c) look to you?

Reply:
would honestly prefer to not use the gnu extension at all, or define(/check?) that nix only supports glibc and musl (both of which can allocate in `getcwd`). doing both can cause (unlikely) odd failures where glibc can keep a very long pwd but other libcs can't :slightly_frowning_face: we can probably just use `std::filesystem`, it's included in at least one place and toml11 seems to use it as well.

Reply:
I'm in the same boat -- I'd prefer `std::filesystem` as well,  but I vaguely recall there being issues in the past. That's why I added the non-`std` stuff on top: to hopefully be in a mergeable state sooner if that was still the case. If `std::filesystem` *is* fine to use, then it's extremely easy to drop my commits and continue off of aszlig's work.